### PR TITLE
[stable] 2 little C++ interface fixes, and one addition for LDC

### PR DESCRIPTION
**Pull request by kinke**

### CODE REVIEW

File: compiler/src/dmd/mtype.h [hunk 1]
```diff
@@ -254,7 +254,6 @@ class Type : public ASTNode
     bool isSharedWild() const  { return (mod & (MODshared | MODwild)) == (MODshared | MODwild); }
     bool isNaked() const       { return mod == 0; }
     Type *nullAttributes() const;
-    Type *arrayOf();
     Type *sarrayOf(dinteger_t dim);
```
Comment (ibuclaw) [inline compiler/src/dmd/mtype.h:257]: This one is missing too.

File: compiler/src/dmd/expression.h [hunk 2]
```diff
@@ -52,6 +53,8 @@ namespace dmd
     Expression *expressionSemantic(Expression *e, Scope *sc);
     // in typesem.d
     Expression *defaultInit(Type *mt, const Loc &loc, const bool isCfile = false);
+    // in enumsem.d
+    Expression *getDefaultValue(EnumDeclaration *ed, const Loc &loc);
```
Comment (ibuclaw) [inline compiler/src/dmd/expression.h:57]: [enum.h](https://github.com/dlang/dmd/blob/stable/compiler/src/dmd/enum.h#L20) seems like a more appropriate location for this declaration.

Comment (ibuclaw) [reply compiler/src/dmd/expression.h:57]: Though I do see why it would be put here as well.

**Review by thewilsonator — APPROVED** [review]

File: compiler/src/dmd/enum.h [hunk 3]
```diff
@@ -17,6 +17,12 @@ class Identifier;
 class Type;
 class Expression;
 
+namespace dmd
+{
+    // in enumsem.d
+    Expression *getDefaultValue(EnumDeclaration *ed, const Loc &loc);
```
Comment (ibuclaw) [inline compiler/src/dmd/enum.h:23]: I hope there's a forward declaration in one of the headers. :-)

**Pull request merged**